database/factories/ModelFactory.php

```diff
@@ -53,6 +53,35 @@
     ];
 });
 
+// @TODO: These should all extend a photo-less "base" post, instead.
```

Comment:
@DFurnes I've been curious about how this would look, would you be able to provide some pseudo-code? I'm thinking it'd help DRY the cleanup work I've been thinking about, for only [setting a Post school ID if its action has its `collect_school_id` field set to `true`.](https://github.com/DoSomething/rogue/blob/d77c9b879e168deaf5118962c2da4fabcb2c4d75/database/factories/ModelFactory.php#L87)

Reply:
Oooh, this was so long ago... 🙈 

I think my thought process here was that we don't need an image or `url` on the `text` posts, but since this "extends" the base `Post` factory above, it's still doing the work to generate a fake image for them anyways. So we're sorta making a photo post and then "undoing" it in the `text` version.

Instead, we could maybe have a base `Post` factory with the common elements (IDs, location, source, whatnot) and then a specific `$factory->defineAs(Post::class, 'photo', …` that adds the photo-specific fields).

Reply:
Gotcha, was thinking it'd be using the post type as a [state](https://laravel.com/docs/5.8/database-testing#factory-states) like you mention, but wanted to be sure. So then if we wanted different [states for the post status as we do currently,](https://github.com/DoSomething/rogue/blob/d77c9b879e168deaf5118962c2da4fabcb2c4d75/database/factories/ModelFactory.php#L94) we could potentially have states like `text-accepted`, `text-rejected`, `photo-accepted` ?

Reply:
Yep, I think that's how we'd have to handle that.